bot/exts/utils/subscription.py

```diff
@@ -0,0 +1,176 @@
+
+        role_check_dict = {
+            roles[role].name: role_check[role] for role in range(len(roles))
+        }
+
+        role_lst = []
+        for role in list(role_check_dict.keys()):
+            if role_check_dict[role]:
+                role_lst.append(role)
+        if len(role_lst) != 0:
+            msg = (
+                f"{', '.join(role_lst[:-1])} and {role_lst[-1]}"
+                if len(role_lst) > 1
+                else role_lst[0]
+            )
+            embed = Embed(
+                title=f"{Emojis.confirmation_emoji} Subscribed",
+                description=f"You've subscribed to {ctx.guild}'s {msg}.",
+                color=Colours.green,
+            )
+        else:
+            embed = Embed(
+                title=f"{Emojis.warning_emoji} Already subscribed",
+                description=f"You're already subscribed to {ctx.guild}'s announcements and polls.",
+                color=Colours.soft_red,
+            )
+        await ctx.send(content=ctx.author.mention, embed=embed)
+
+    async def unsubscribe_group_helper(self, ctx: Context) -> None:
+        """Helper function for unsubscribe_group."""
+        roles = (
+            ctx.guild.get_role(Roles.announcements),
+            ctx.guild.get_role(Roles.polls),
+        )
+        role_check = ()
+
+        for i in roles:
+            role_check += (await self.de_apply_role(ctx, i),)
+
+        role_check_dict = {
+            roles[role].name: role_check[role] for role in range(len(roles))
+        }
+
+        role_lst = []
+        for role in list(role_check_dict.keys()):
+            if role_check_dict[role]:
+                role_lst.append(role)
+        if len(role_lst) != 0:
+            msg = (
+                f"{', '.join(role_lst[:-1])} and {role_lst[-1]}"
+                if len(role_lst) > 1
+                else role_lst[0]
+            )
+            embed = Embed(
+                title=f"{Emojis.confirmation_emoji} Unsubscribed",
+                description=f"You've unsubscribed to {ctx.guild}'s {msg}.",
+                color=Colours.green,
+            )
+        else:
+            embed = Embed(
+                title=f"{Emojis.warning_emoji} Already unsubscribed",
+                description=f"You're already unsubscribed to {ctx.guild}'s announcements and polls.",
+                color=Colours.soft_red,
+            )
+        await ctx.send(content=ctx.author.mention, embed=embed)
+
+    @commands.group(name="subscribe", invoke_without_command=True)
+    async def subscribe_group(self, ctx: Context) -> None:
+        """Subscribe to announcements and polls notifications by assigning yourself the role."""
+        await self.subscribe_group_helper(ctx)
+
+    @commands.group(name="unsubscribe", invoke_without_command=True)
+    async def unsubscribe_group(self, ctx: Context) -> None:
+        """Unsubscribe to announcements and polls notifications by assigning yourself the role."""
```

Comment:
This is probably a comma splice and there are two roles so `role` should be plural.
```suggestion
        """Unsubscribe to announcements and polls notifications, by assigning yourself the roles."""
```
Mention this change in other similar places too.